src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -1005,7 +1009,7 @@ internal bool VerifyRemoteCertificate(RemoteCertValidationCallback remoteCertVal
                 if (remoteCertificateEx == null)
                 {
                     if (NetEventSource.IsEnabled)
-                        NetEventSource.Exit(this, "(no remote cert)", !_sslAuthenticationOptions.RemoteCertRequired);
+                        NetEventSource.Exit(this, $"No remote certificate received. RemoteCertRequired: {RemoteCertRequired}");
```

Comment:
why is this changing from `!_sslAuthenticationOptions.RemoteCertRequired` to `RemoteCertRequired`?  Is it an equivalent value?  It looks like it might be missing the "!" operator ?

Reply:
I changed it because I found the log message is confusing, and I want to improve it.

The `VerifyRemoteCertificate()` function tries to get the remote certificate (the certificate from other side). When there is no remote certificate received, it makes more sense to log whether current host (client/server) requires remote certificate or not.

```c#
        internal bool RemoteCertRequired
        {
            get
            {
                return _sslAuthenticationOptions.RemoteCertRequired;
            }
        }
```

I apply both changes just to compare the output message side-by-side.
<img width="724" alt="capture" src="https://user-images.githubusercontent.com/8537784/43018233-d4b5ad76-8c0d-11e8-940d-3750f9eb35d0.PNG">

Reply:
>When there is no remote certificate received, it makes more sense to log whether current host (client/server) requires remote certificate or not.

I don't agree with this completely.  There is a difference in whether the "local" side is requesting a certificate or not from the "remote" side vs. whether the "remote" side provided one at all.  In fact, it probably makes sense to log both pieces of information.  Most connection failures are because the "local" side requested a cert and the "remote" side did not send one.

Reply:
I think current modified logging include both pieces of information.

> whether the "local" side is requesting a certificate or not from the "remote" side

This information is in `RemoteCertRequired`.

> whether the "remote" side provided one at all

This information is in `No remote certificate received`.

Did I misunderstand something?